nntrainer/graph/network_graph.cpp

```diff
@@ -344,10 +344,21 @@ void NetworkGraph::applyGradients(
   }
 }
 
-sharedConstTensors
-NetworkGraph::forwarding(bool training,
-                         std::function<bool(void *userdata)> stop_cb) {
-  for (auto iter = cbegin(); iter != cend() && !stop_cb(nullptr); iter++) {
+sharedConstTensors NetworkGraph::forwarding(
+  ml::train::RunStats &stat, bool training,
+  std::function<bool(void *userdata)> stop_cb,
+  ml::train::StopCallbackCheckpointType stop_cb_checkpoint, void *user_data) {
+  for (auto iter = cbegin(); iter != cend(); iter++) {
+    if (enum_class_logical_and(
+          stop_cb_checkpoint,
+          ml::train::StopCallbackCheckpointType::EVERY_FORWARDING_LAYER)) {
```

Comment:
It might be better user_data to have this StopCallbackCheckpointType and guide users on how to use it.
Otherwise, the API is too messy.